lib/mosaic-database/src/main/java/org/eclipse/mosaic/lib/database/route/Route.java

```diff
@@ -99,11 +79,19 @@ public void addEdge(@Nonnull Edge edge) {
      */
     @Nonnull
     public List<Node> getNodes() {
-        if (edgeList.isEmpty()) {
+        if (connections.isEmpty()) {
             return new ArrayList<>();
         } else {
-            List<Node> nodes = edgeList.stream().map(Edge::getFromNode).collect(Collectors.toList());
-            nodes.add(edgeList.get(edgeList.size() - 1).getToNode());
+            // we store all nodes of all connections here
+            // this is needed for determining the next and previous node in LazyLoadingRoadPosition
+            List<Node> nodes = new ArrayList<>();
+            for (Connection connection : connections) {
```

Comment:
I feel like it would be cleaner to store the `Node`s in `Connection` as tuple, since there are always 2 and then only call `nodes.add(connection.getNodes.getA()`, though this would require extra handling for the last connection, since we also have to add the last node

Reply:
Connections can contain any arbitrary number of nodes, but at least two.

Reply:
Or I guess even better use `connection.getFrom()` and `connection.getTo()`

Reply:
It is required that the list of nodes contains **all** nodes of all connections, not only the from/to node.

Reply:
Ah okay, I thought `Connection`s would just have two `Node`s after the change. But we are keeping the "in-between Nodes"?

Reply:
yes, we need them for the shape of the connections. we just skip transforming them into several edges.